js.js

```diff
@@ -72,8 +73,46 @@ module.exports = {
     // parse various forms correctly. For now warn on invalid type froms,
     // should revisit once following issue is fixed:
     // https://github.com/jsdoctypeparser/jsdoctypeparser/issues/50
-    'jsdoc/valid-types': 'off'
-
+    'jsdoc/valid-types': 'off',
+    'import/order': [
+      'error',
+      {
+        alphabetize: {
+          order: 'asc',
+          caseInsensitive: false
+        },
+        'newlines-between': 'always',
+        // the overall order of imports
+        groups: ['builtin', 'external', 'internal', ['sibling', 'parent', 'index'], 'object'],
+        pathGroupsExcludedImportTypes: ['builtin'],
+        pathGroups: [
+          { // 3p imports
+            pattern: '+(!(@))*/**',
+            group: 'external',
+            position: 'before',
+            patternOptions: { nonegate: true }
+          },
+          { // scoped external imports excluding our 2p scoped packages
+            pattern: '@!(ipfs|libp2p|ipfs-shipyard|ipld|multiformats)/**',
+            group: 'external',
+            position: 'after',
+            patternOptions: { nonegate: false }
+          },
+          { // our 2p scoped packages
+            pattern: '@{ipfs,libp2p,ipfs-shipyard,ipld,multiformats}/**',
+            group: 'parent',
+            position: 'before',
+            patternOptions: { nonegate: true }
+          },
+          { // our relative (this package) imports
+            pattern: '.*/**',
+            group: 'index',
+            position: 'after',
+            patternOptions: { dot: true, nonegate: true }
+          }
+        ]
+      }
+    ]
```

Comment:
How about a bit simpler?  Built-ins, externals, types then everything else.

This way the externals are sorted in the same order as the deps in `package.json`.

The only outlier here I think is "object imports" but I'm not sure we use them anywhere.

```suggestion
    'import/order': [
      'error',
      {
        alphabetize: {
          order: 'asc',
          caseInsensitive: false
        },
        'newlines-between': 'always',
        // the overall order of imports - anything not in this list is grouped together at the end
        groups: ['builtin', 'external', 'type']
      }
    ]
```